sql/moz-fx-data-shared-prod/firefox_ios_derived/firefox_ios_clients_v1/query.sql

```diff
@@ -0,0 +1,285 @@
+-- activations AS (
+--   SELECT
+--     client_id,
+--     ARRAY_AGG(activated ORDER BY submission_date DESC)[SAFE_OFFSET(0)] > 0 AS activated
+--   FROM
+--     `moz-fx-data-shared-prod.fenix.new_profile_activation`
+--   WHERE
+--     submission_date = @submission_date
+--   GROUP BY
+--     client_id
+-- ),
+-- Find earliest data per client from the first_session ping.
+first_session_ping AS (
+  SELECT
+    client_id,
+    MIN(sample_id) AS sample_id,
+    DATETIME(MIN(submission_timestamp)) AS min_submission_datetime,
+    DATE(MIN(SAFE.PARSE_DATETIME('%F', SUBSTR(first_run_date, 1, 10)))) AS first_run_date,
+    ARRAY_AGG(
+      IF(
+        -- we should probably do the nullif prior to doing this is not null check
+        adjust_ad_group IS NOT NULL
+        OR adjust_campaign IS NOT NULL
+        OR adjust_creative IS NOT NULL
+        OR adjust_network IS NOT NULL,
+        STRUCT(
+          submission_timestamp,
+          adjust_ad_group,
+          adjust_campaign,
+          adjust_creative,
+          adjust_network
+        ),
+        NULL
+      ) IGNORE NULLS
+      ORDER BY
+        submission_timestamp
+      LIMIT
+        1
+    )[SAFE_OFFSET(0)] AS adjust_info,
+  FROM
+    (
+      SELECT
+        client_info.client_id,
+        sample_id,
+        submission_timestamp,
+        client_info.first_run_date,
+        NULLIF(metrics.string.adjust_ad_group, "") AS adjust_ad_group,
+        NULLIF(metrics.string.adjust_campaign, "") AS adjust_campaign,
+        NULLIF(metrics.string.adjust_creative, "") AS adjust_creative,
+        NULLIF(metrics.string.adjust_network, "") AS adjust_network,
+      FROM
+        `moz-fx-data-shared-prod.firefox_ios.first_session`
+      WHERE
+        DATE(submission_timestamp) = @submission_date
+        AND client_info.client_id IS NOT NULL
+        AND ping_info.seq = 0 -- Pings are sent in sequence, this guarantees that the first one is returned.
+    )
+  GROUP BY
+    client_id
+),
+-- Find earliest data per client from the metrics ping.
+metrics_ping AS (
+  SELECT
+    client_id,
+    MIN(sample_id) AS sample_id,
+    DATETIME(
+      MIN(submission_timestamp)
+    ) AS min_submission_datetime,  -- do we need this? Looks like later we might want to see timestamp corresponding to the row that contains the campaign info
+    ARRAY_AGG(
+      IF(
+        adjust_ad_group IS NOT NULL
+        OR adjust_campaign IS NOT NULL
+        OR adjust_creative IS NOT NULL
+        OR adjust_network IS NOT NULL,
+        STRUCT(
+          submission_timestamp,
+          adjust_ad_group,
+          adjust_campaign,
+          adjust_creative,
+          adjust_network
+        ),
+        NULL
+      ) IGNORE NULLS
+      ORDER BY
+        submission_timestamp
+      LIMIT
+        1
+    )[SAFE_OFFSET(0)] AS adjust_info,
+    -- # TODO: understand better the install_source field.
+    -- what exactly does it contain?
+    ARRAY_AGG(install_source IGNORE NULLS ORDER BY submission_timestamp ASC)[
```

Comment:
I don't think we have install source for iOS: https://dictionary.telemetry.mozilla.org/apps/firefox_ios?page=1&search=source